src/components/ClaimErrorAlert/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { AlertCard, Button, Images } from '@hyperplay/ui'
+import { useTranslation } from 'react-i18next'
+import { Chain, mantle } from 'viem/chains'
+import { ClaimError, NotEnoughGasError, WarningError } from '@/types/quests'
+import { errorIsSwitchChainError } from '@/helpers/claimErrors'
+
+import styles from './index.module.scss'
+
+const { AlertOctagon, WarningIcon } = Images
+
+const gasInformation: Record<number, { url: string }> = {
+  [mantle.id]: {
+    url: 'https://www.mantle.xyz/mnt'
+  }
+}
+
+type ClaimErrorAlertProps = {
+  error: Error | WarningError | ClaimError
+  networkName: string
+  currentChain: Chain | undefined
+  onOpenDiscordLink: () => void
+}
+
+export const ClaimErrorAlert = ({
+  error,
+  networkName,
+  onOpenDiscordLink,
+  currentChain
+}: ClaimErrorAlertProps) => {
+  const { t } = useTranslation()
+
+  if (error instanceof WarningError) {
+    return (
+      <AlertCard
+        icon={<WarningIcon />}
+        showClose={false}
+        title={error.title}
+        message={error.message}
+        variant="warning"
+        noBorderLeft={true}
+      />
+    )
+  }
+
+  if (error instanceof NotEnoughGasError) {
+    const currency = currentChain?.nativeCurrency.symbol ?? 'Unknown'
+    const gasUrl = currentChain ? gasInformation[currentChain.id]?.url : null
+
+    return (
+      <AlertCard
+        icon={<AlertOctagon />}
+        showClose={false}
+        title={t('quest.notEnoughGas.title', `Not enough {{symbol}}`, {
+          symbol: currency
+        })}
+        message={
+          <>
+            <div>
+              {t(
+                'quest.notEnoughGas.message',
+                `You'll need a bit of {{symbol}} to claim your reward.`,
+                {
+                  symbol: currency
+                }
+              )}
+            </div>
+            {gasUrl ? (
+              <div>
+                <Button type="link" className={styles.link}>
+                  <a href={gasUrl} target="_blank" rel="noopener noreferrer">
+                    {t('quest.notEnoughGas.clickHere', `Click here`)}
+                  </a>
+                </Button>{' '}
+                {t(
+                  'quest.notEnoughGas.thenComeBack',
+                  `to learn how to get some, then come back to try again.`
+                )}
+              </div>
+            ) : null}
```

Comment:
I think you could clean this up a bit with the `Trans` component https://react.i18next.com/latest/trans-component#using-with-react-components

What do you think?

Reply:
Tried doing this but it still wasn't rendering the link properly

```tsx
<Trans
  i18nKey="quest.notEnoughGas.getGas"
  defaults="<link>Click here</link> to learn how to get some, then come back to try again."
  components={{
    link: (
      <Button
        type="link"
        className={styles.link}
      >
        <a
          href={gasUrl}
          target="_blank"
          rel="noopener noreferrer"
        />
      </Button>
    )
  }}
/>
```